### PR TITLE
Remove synthetic unwinder error frames from agent and instead generate them in the unwinder with more context

### PR DESCRIPTION
**Pull request by gnurizen**
The main benefit here is getting line numbers from the errors, there is roughly a couple dozen error sites so having line numbers is nice.  The symbol "class" is the error message, a little clunky but works.  Additionally we add the samples from exitted processes now instead of throwing them away.

### CODE REVIEW

File: pkg/pprof/pprof.go [hunk 1]
```diff
@@ -437,9 +394,20 @@ func (c *Converter) AddUnwinderInfoLocation(frameID uint64) *pprofprofile.Locati
 		return l
 	}
 
+	mapping := c.interpreterMapping
+	for idx, prog := range bpfprograms.ProgNames {
+		if base, found := strings.CutSuffix(interpreterSymbol.Filename, ".c"); found {
+			_, file := path.Split(base)
+			if strings.HasSuffix(prog, file+".o") {
+				mapping = c.bpfProgMappings[idx]
```
Comment (gnurizen) [inline pkg/pprof/pprof.go:402]: This is kinda hacky, open to suggestions.

File: bpf/unwinders/native.bpf.c [hunk 2]
```diff
@@ -948,6 +948,7 @@ int native_unwind(struct bpf_perf_event_data *ctx) {
             LOG("[warn] mapping not found");
             request_refresh_process_info(ctx, per_process_id);
             BUMP_UNWIND_FAILED_COUNT(per_process_id, mapping_not_found);
```
Comment (umanwizard) [inline bpf/unwinders/native.bpf.c:950]: We can probably get rid of `BUMP_UNWIND_FAILED_COUNT`, `failed_unwind_reasons`, etc. now

File: pkg/pprof/pprof.go (hunk 1 above)
Comment (umanwizard) [reply pkg/pprof/pprof.go:407]: We use `StackFrame.Status` in `profile.go` to track truncated frames today. You could add a new possible value for bpf error frames, then pass it through to here.

File: bpf/unwinders/native.bpf.c (hunk 2 above)
Comment (gnurizen) [reply bpf/unwinders/native.bpf.c:950]: I was thinking these could still be useful for surveillance purposes (ie if we got them exported into prometheus).

Comment (umanwizard) [reply bpf/unwinders/native.bpf.c:950]: We already have a bunch of Prometheus metrics from the unwinder, which doesn't work via this `BUMP_UNWIND_FAILED_COUNT` mechanism. See e.g. `unwinder_stats_t` and the `DEFINE_COUNTER` mechanism.

**Review by umanwizard — APPROVED** [review]

**Pull request merged**